ding/policy/d4pg.py

```diff
@@ -204,13 +227,35 @@ def _init_learn(self) -> None:
         self._forward_learn_cnt = 0  # count iterations
 
     def _forward_learn(self, data: dict) -> Dict[str, Any]:
-        r"""
+        """
         Overview:
-            Forward and backward function of learn mode.
+            Policy forward function of learn mode (training policy and updating parameters). Forward means \
+            that the policy inputs some training batch data from the replay buffer and then returns the output \
+            result, including various training information such as different loss, actor and critic lr.
         Arguments:
-            - data (:obj:`dict`): Dict type data, including at least ['obs', 'action', 'reward', 'next_obs']
+            - data (:obj:`dict`): Input data used for policy forward, including the \
+                collected training samples from replay buffer. For each element in dict, the key of the \
+                dict is the name of data items and the value is the corresponding data. Usually, the value is \
+                torch.Tensor or np.ndarray or there dict/list combinations. In the ``_forward_learn`` method, data \
+                often need to first be stacked in the batch dimension by some utility functions such as \
+                ``default_preprocess_learn``. \
+                For D4PG, each element in list is a dict containing at least the following keys: ``obs``, \
+                ``action``, ``reward``, ``next_obs``. Sometimes, it also contains other keys such as ``weight``.
+
         Returns:
-            - info_dict (:obj:`Dict[str, Any]`): Including at least actor and critic lr, different losses.
+            - info_dict (:obj:`Dict[str, Any]`): The output result dict of forward learn, \
+                containing at least the "actor lr", "critic lr", "different losses", "q_value", "action", "priority", \
+                keys. Additionally, loss_dict also contain other keys, which are mainly used for monitoring and \
```

Comment:
"Additionally, loss_dict also contain other keys, which are mainly used for monitoring and" -> "Additionally, loss_dict also contains other keys, which is mainly used for monitoring and"